# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/WindowGlobalParent.h
# Commit: 8d7f44b9b6e2
# Full Hash: 8d7f44b9b6e2debb08256460bc77355551e5d3c8
# Author: Paul Zuehlcke <pbz@mozilla.com>
# Date: 2023-09-19 03:53:26
# Regressor Bug: 1839918
# File Overlap Count: 3
# Description:
#   Bug 1839918 - Create SetCookies variant for document.cookie on PWindowGlobal. r=bvandersloot,cookie-reviewers,valentin
#   
#   Using PWindowGlobal for SetCookies calls from content to parent process means we can get
#   the BrowsingContext the cookies are set for. The current implementation in PCookieService
#   drops this relationship.
# ==============================================================================

diff -r 7a28b63b4fc3 -r 8d7f44b9b6e2 dom/ipc/WindowGlobalParent.h
--- a/dom/ipc/WindowGlobalParent.h	Mon Sep 18 11:55:09 2023 +0000
+++ b/dom/ipc/WindowGlobalParent.h	Mon Sep 18 11:55:09 2023 +0000
@@ -313,6 +313,10 @@
   mozilla::ipc::IPCResult RecvHasStorageAccessPermission(
       HasStorageAccessPermissionResolver&& aResolve);
 
+  mozilla::ipc::IPCResult RecvSetCookies(
+      const nsCString& aBaseDomain, const OriginAttributes& aOriginAttributes,
+      nsIURI* aHost, bool aFromHttp, const nsTArray<CookieStruct>& aCookies);
+
  private:
   WindowGlobalParent(CanonicalBrowsingContext* aBrowsingContext,
                      uint64_t aInnerWindowId, uint64_t aOuterWindowId,